setup/command/apply.go

```diff
@@ -16,15 +16,24 @@ func (f *CommandFactory) Apply() cli.Command {
 				Name:  "quick",
 				Usage: "skips verification checks that normally run after 'terraform apply' has completed",
 			},
+			cli.BoolTFlag{
+				Name:  "push",
+				Usage: "setting it to false skips pushing local tfstate to s3 (default: true)",
+			},
 		},
 		Action: func(c *cli.Context) error {
 			args, err := extractArgs(c.Args(), "NAME")
 			if err != nil {
 				return err
 			}
 
+			provider, err := f.newAWSProviderHelper(c)
```

Comment:
So as I understand it, the current code path is, if the awsFlags aren't explicitly passed in, we would read fall back on `aws.Config.Credentials`which searches for credentials in env variables, credential files etc.

- Is this the desired behavior?
- If so, can we consider UX acceptable?